javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ConfigurationFileLoader.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.config;
+
+import static java.util.Collections.emptyMap;
+import static java.util.logging.Level.SEVERE;
+
+import io.opentelemetry.instrumentation.api.internal.ConfigPropertiesUtil;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public final class ConfigurationFileLoader implements Supplier<Map<String, String>> {
+
+  private static final Logger logger = Logger.getLogger(ConfigurationFileLoader.class.getName());
+
+  static final String CONFIGURATION_FILE_PROPERTY = "otel.javaagent.configuration-file";
+
+  @Override
+  public Map<String, String> get() {
```

Comment:
There's a fair amount going on in this method. I think it would be helpful to have some test coverage...but the current structure is a bit harder to test due to the static calls to `ConfigPropertiesUtil.getString()` and the inline creation of file streams. Still, there are at least 5 untested branches....

Reply:
It is actually tested, in `ConfigurationFileTest`.